android/sdl_android/src/androidTest/java/com/smartdevicelink/managers/screen/SoftButtonManagerTests.java

```diff
@@ -402,4 +407,55 @@ public void testSoftButtonStateEquals() {
         softButtonState2 = new SoftButtonState("object1-state1", "o1s1", artwork1);
         assertEquals(softButtonState1, softButtonState2);
     }
+
+    /**
+     * Test constructing SoftButtonObject with an empty state list
+     */
+    @Test
+    public void testConstructSoftButtonObjectWithEmptyStateList() {
+        List<SoftButtonState> stateList = new ArrayList<>();
+        SoftButtonObject softButtonObject1 = new SoftButtonObject("hello_there", stateList, "general_kenobi", null);
+        assertNull(softButtonObject1.getStates());
+    }
+
+    /**
+     * Test constructing SoftButtonObject with an nonempty state list
+     */
+    @Test
+    public void testConstructingSoftButtonObjectWithNonEmptyStateList() {
+        List<SoftButtonState> stateList = new ArrayList<>();
+        SoftButtonState softButtonState = new SoftButtonState("general_kenobi", "General Kenobi", null);
+        stateList.add(softButtonState);
+        SoftButtonObject softButtonObject2 = new SoftButtonObject("hello_there_again", stateList, "general_kenobi", null);
+        assertEquals(stateList, softButtonObject2.getStates());
+    }
+
+    /**
+     * Test assigning an empty state list to existing SoftButtonObject
+     */
+    @Test
+    public void testAssignEmptyStateListToSoftButtonObject() {
+        List<SoftButtonState> stateList = new ArrayList<>();
+        SoftButtonState softButtonState = new SoftButtonState("object1-state1", "o1s1", null);
+
+        SoftButtonObject softButtonObject = new SoftButtonObject("hi", softButtonState, null);
+
+        softButtonObject.setStates(stateList);
+        assertNotEquals(stateList, softButtonObject.getStates());
```

Comment:
Rather than assert that these are not equal we could assert that the state on the Object is left unchanged after calling setStates.
`assertEquals(softButtonState, softButtonObject.getStates)`
You may want to use the constructor that takes a list for states rather than a single state. or you could make a list out of `softButtonState` before you make the assertion